### ISSUE TITLE
refilled lamp with lamp oil, but already containing some lamp oil (3units)

### ISSUE THREAD

**Issue by Shadesmar99** [CleverRaven/Cataclysm-DDA#55963]
### Describe the bug

as title stated. received bug report. Lamp oil i had is gone now. (the 503 units referred to in the bug report)

### Steps to reproduce

as title states

### Expected behavior

lamp refills normally (because lamp oil should be able to mix with lamp oil) and lamp oil doesnt vanish

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19042.1526 (20H2)
- Game Version: 2c922b0 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm],
    Stats Through Kills [stats_through_kills]
]

### Additional context

 DEBUG    : tried to put an item (lamp_oil) count (503) in a container (oil_lamp) that cannot contain it: pocket unacceptable because can't mix liquid with contained item

 FUNCTION : ret_val<bool> item::put_in(const item&, item_pocket::pocket_type, bool)
 FILE     : src/item.cpp
 LINE     : 1405
 VERSION  : 2c922b0

**Comment by Maleclypse** [comment]
Does lamp oil have an expiration date?  If so that's why?  If not then no clue what's happening.

**Comment by DrMorbius42** [comment]
Just an FYI.  I had this issue today and it seems to be a compatibility issue with fuel made from a previous version.  I made some "crude lamp oil" from several experimental versions back (not sure which one) and I could not refill a crude oil lamp getting the same message above if there was fuel in the lamp.  I resolved the issue by discarding the existing crude lamp oil I made from previous versions and used a new batch I made with the current experimental (v2022-11-20-1651).  Once I used the fuel made from the same version the lamp refilled properly.
